File: retaguarda-software-house/java/retaguarda_sh/src/main/java/com/t2ti/retaguarda_sh/model/transiente/Filtro.java
package com.t2ti.retaguarda_sh.model.transiente;

import java.io.Serializable;

/**
 * Classe transiente para controlar o filtro que vem do cliente.
 * tomar como base o seguinte esquema para criar as condições de filtro (filter conditions)
 * https://github.com/nestjsx/crud/wiki/Requests
 */
public class Filtro implements Serializable {
	private static final long serialVersionUID = 1L;

	private String campo;
	private String valor;
	private String dataInicial;
	private String dataFinal;
	private String where;
	private String condicao;

	public Filtro() {}
	
	public Filtro(String filter) {
		String filtro = filter.replace("||", ":");
		String[] partesDoFiltro = filtro.split("\\?");
		
		for (int i = 0; i < partesDoFiltro.length; i++) {
			String[] condicoes = partesDoFiltro[i].split(":");

	        setCondicao(condicoes[1]);

	        if (i > 0) {
	            setWhere(getWhere() + " AND ");
	        }
	        
	        // $cont (LIKE %val%, contains)
	        if (condicao.equals("$cont")) {
	            this.campo = condicoes[0];
	            this.valor = condicoes[2];
	            setWhere(getWhere() + campo + " like '%" + valor + "%'");
	        }

	        // $eq (=, equal)
	        if (condicao.equals("$eq")) {
	            this.campo = condicoes[0];
	            this.valor = condicoes[2];
	            setWhere(getWhere() + campo + " = '" + valor + "'");
	        }
	        
	        // $between (BETWEEN, between, accepts two values)
	        if (condicao.equals("$between")) {
				String[] datas = condicoes[2].split(",");
	            this.campo = condicoes[0];
	            this.dataInicial = datas[0];
	            this.dataFinal = datas[1];
	            setWhere(getWhere() + campo + " between '" + dataInicial + "' and '" + dataFinal + "'");
	        }			
		}
		
        
	}

	public String getCampo() {
		return campo;
	}

	public void setCampo(String campo) {
		this.campo = campo;
	}

	public String getValor() {
		return valor;
	}

	public void setValor(String valor) {
		this.valor = valor;
	}

	public String getDataInicial() {
		return dataInicial;
	}

	public void setDataInicial(String dataInicial) {
		this.dataInicial = dataInicial;
	}

	public String getDataFinal() {
		return dataFinal;
	}

	public void setDataFinal(String dataFinal) {
		this.dataFinal = dataFinal;
	}

	public String getWhere() {
		if (where == null)
			return "";
	    return where;
	}

	public void setWhere(String where) {
		this.where = where;
	}

	public String getCondicao() {
		return condicao;
	}

	public void setCondicao(String condicao) {
		this.condicao = condicao;
	}

}

File: retaguarda-software-house/c-sharp/T2TiRetaguardaSH/T2TiRetaguardaSH/Models/Transiente/Filtro.cs
/**
 * Classe transiente para controlar o filtro que vem do cliente.
 * tomar como base o seguinte esquema para criar as condições de filtro (filter conditions)
 * https://github.com/nestjsx/crud/wiki/Requests
 */
namespace T2TiRetaguardaSH.Models
{
    public class Filtro
    {

        public Filtro() { }

        public Filtro(string filter)
        {
            string[] partesDoFiltro = filter.Split("?");

            for (int i = 0; i < partesDoFiltro.Length; i++)
            {
                string[] condicoes = partesDoFiltro[i].Split("||");

                if (i > 0)
                {
                    Where = Where + " AND ";
                }

                // $cont (LIKE %val%, contains)
                if (condicoes[1] == "$cont")
                {
                    Campo = condicoes[0];
                    Valor = condicoes[2];
                    Where = Where + Campo + " like '%" + Valor + "%'";
                }

                // $eq (=, equal)
                if (condicoes[1] == "$eq")
                {
                    Campo = condicoes[0];
                    Valor = condicoes[2];
                    Where = Where + Campo + " = " + Valor + "'";
                }

                // $between (BETWEEN 'DATA1' and 'DATA2')
                if (condicoes[1] == "$between")
                {
                    string[] datas = condicoes[2].Split(",");
                    Campo = condicoes[0];
                    DataInicial = datas[0];
                    DataFinal = datas[1];
                    Where = Where + Campo + " between '" + DataInicial + "' and '" + DataFinal + "'";
                }
            }

        }

        public string Campo { get; set; }
        public string Valor { get; set; }
        public string DataInicial { get; set; }
        public string DataFinal { get; set; }

        public string Where { get; set; }
    }
}
